window/runner_panel.py

```diff
@@ -236,3 +252,23 @@ def finished(self):
                 self.process_killed = False
             else:
                 self.message("Subprocess completed.")
+                if "steamcmd" in self.process.program():
+                    if self.steamcmd_failed_mods != []:
+                        tempdata = "Failed to connect"  # Default value
+                        GetPublishedFileDetails = (
```

Comment:
This relies on there being an internet connection, and if a mod failed because no internet connection... we get crash. I will add sanitization to avoid the crash, but additional handling would still need done, and since we don't really need to display the names to the user to prompt them to redownload, instead of doing this warning, we could possibly instead do a prompt to redownload or retry  button for the failed operation?

I think that'd be a great compromise, and would actually be even more convenient for the user. You could just use `model.dialogue.show_dialogue_conditional` to prompt a No/Yes prompt like we do for the "Setup SteamCMD" stuffs when it prompts to reinstall or do the symlink and have it say:

"X mod(s) failed to download. Do you want to try to redownload these mods?"

And then skip the lookup with WebAPI because you basically have to just iterate through the downloader_list and remove the mods that succeeeded, based on the pfids added to your list `self.steamcmd_failed_mods`

We should also be doing a `runner.message(self.steamcmd_failed_mods)` of the failed mods so that the user can retrieve the pfids from the `RunnerPanel` if push comes to shove

Reply:
In addition to the above, please also add an effect to the Progress Bar that indicates that not all mods failed. I think that changing the color of the progress to a yellow or red would suffice. The percentage should also not add up to 100% if a mod failed. It should only show the percentage for mods that were successfully downloaded out of the total mods being downloaded.